nexus/src/app/oximeter.rs

```diff
@@ -212,14 +216,82 @@ impl super::Nexus {
             .map_err(|e| Error::internal_error(&e.to_string()))?
             .timeseries_schema_list(&pag_params.page, limit)
             .await
-            .map_err(|e| match e {
-                oximeter_db::Error::DatabaseUnavailable(_) => {
-                    Error::ServiceUnavailable {
-                        internal_message: e.to_string(),
-                    }
-                }
-                _ => Error::InternalError { internal_message: e.to_string() },
-            })
+            .map_err(map_oximeter_err)
+    }
+
+    pub async fn select_timeseries(
+        &self,
+        timeseries_name: &str,
+        criteria: &[&str],
+        interval: Duration,
```

Comment:
I agree, it would be better to accept and query a start/end time, which is the API in `oximeter_db` that's being used inside this call. But if accepting the interval ends up being useful, Nexus does have access to it in the `metric_producer` table, which has an `interval` column.

Reply:
I think I'm gonna try modifying [oximeter's `select_timeseries_with`](https://github.com/oxidecomputer/omicron/blob/947449c4069234d8f6da3a9ef9b209fa26cfac6e/oximeter/db/src/client.rs#L63) function to accept a limit parameter, and use that instead of the `interval`.

You're right, it is useful for Nexus to know the `interval` when querying, but IMO, that shouldn't be necessary when `SELECT`ing.

Reply:
Yeah, that sounds good. The [query builder](https://github.com/oxidecomputer/omicron/blob/947449c4069234d8f6da3a9ef9b209fa26cfac6e/oximeter/db/src/query.rs#L34) already supports limit/offset pagination, I just never wired it up to the public API since there were no consumers at the time.